access_requests.go

```diff
@@ -0,0 +1,240 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// AccessRequest represents a access request for a group or project.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/access_requests.html
+// TODO: try to find existed structure for that
+type AccessRequest struct {
+	ID        int        `json:"id"`
+	Username  string     `json:"username"`
+	Name      string     `json:"name"`
+	State     string     `json:"state"`
+	CreatedAt *time.Time `json:"created_at"`
+	// RequestedAt represents a access request requested time. Can be empty, when used
+	// ApproveProjectAccessRequest() or ApproveGroupAccessRequest().
+	RequestedAt *time.Time `json:"requested_at"`
+	// AccessLevel represents a user access level. Can be empty, when used ListProjectAccessRequests(),
+	// ListGroupAccessRequests(), RequestProjectAccess(), RequestGroupAccess().
+	AccessLevel *AccessLevelValue `json:"access_level,omitempty"`
+}
+
+// AccessRequestsService handles communication with the project/group access requests
+// related methods of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/access_requests.html
+type AccessRequestsService struct {
+	client *Client
+}
+
+// ListAccessRequestsOptions represents the available ListProjectAccessRequests() or ListGroupAccessRequests()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/access_requests.html#list-access-requests-for-a-group-or-project
+type ListAccessRequestsOptions ListOptions
+
+// ListProjectAccessRequests gets a list of access requests
+// viewable by the authenticated user.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/access_requests.html#list-access-requests-for-a-group-or-project
+func (s *AccessRequestsService) ListProjectAccessRequests(pid interface{}, opt *ListAccessRequestsOptions, options ...OptionFunc) ([]*AccessRequest, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/access_requests", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ar []*AccessRequest
```

Comment:
In list calls we try to add an `s` so the variable names indicates it are multiple instead of a single `ar` (goes for all cases in this file of course).